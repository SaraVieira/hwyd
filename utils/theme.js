import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
