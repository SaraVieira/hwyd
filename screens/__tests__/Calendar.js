import React from 'react';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import { Calendar } from '../Calendar';
import theme from '../../utils/theme';

describe('<Calendar />', () => {
  it('renders', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Calendar />
      </ThemeProvider>,
    );

    const root = tree.find(Calendar).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders with markedDates', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <Calendar />
      </ThemeProvider>,
    );

    const root = tree.find(Calendar).shallow();
    root.setState({
      markedDates: {
        '03-02-2018': { disabled: false, marked: true },
        '03-03-2018': { disabled: false, marked: true },
        '03-04-2018': { disabled: false, marked: true },
      },
    });
    expect(root).toMatchSnapshot();
  });
});
