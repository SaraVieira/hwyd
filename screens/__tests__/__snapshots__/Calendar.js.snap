// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> renders 1`] = `
<Styled(View)>
  <Styled(LinearGradient)
    colors={
      Array [
        "#f8a5c2",
        "#f7d794",
      ]
    }
  />
  <CalendarList
    futureScrollRange={0}
    markedDates={Object {}}
    onDayPress={[Function]}
    pastScrollRange={12}
    style={
      Object {
        "paddingTop": 24,
      }
    }
    theme={
      Object {
        "backgroundColor": "transparent",
        "calendarBackground": "transparent",
        "dayTextColor": "#000000",
        "dotColor": "#cf6a87",
        "selectedDayBackgroundColor": "#cf6a87",
        "selectedDayTextColor": "#ffffff",
        "textDayFontFamily": "libre-regular",
        "textDayHeaderFontFamily": "lato-black",
        "textDisabledColor": "#596275",
        "textMonthFontFamily": "libre-bold",
        "textMonthFontSize": 18,
        "textSectionTitleColor": "#cf6a87",
        "todayTextColor": "#cf6a87",
      }
    }
  />
</Styled(View)>
`;

exports[`<Calendar /> renders with markedDates 1`] = `
<Styled(View)>
  <Styled(LinearGradient)
    colors={
      Array [
        "#f8a5c2",
        "#f7d794",
      ]
    }
  />
  <CalendarList
    futureScrollRange={0}
    markedDates={
      Object {
        "03-02-2018": Object {
          "disabled": false,
          "marked": true,
        },
        "03-03-2018": Object {
          "disabled": false,
          "marked": true,
        },
        "03-04-2018": Object {
          "disabled": false,
          "marked": true,
        },
      }
    }
    onDayPress={[Function]}
    pastScrollRange={12}
    style={
      Object {
        "paddingTop": 24,
      }
    }
    theme={
      Object {
        "backgroundColor": "transparent",
        "calendarBackground": "transparent",
        "dayTextColor": "#000000",
        "dotColor": "#cf6a87",
        "selectedDayBackgroundColor": "#cf6a87",
        "selectedDayTextColor": "#ffffff",
        "textDayFontFamily": "libre-regular",
        "textDayHeaderFontFamily": "lato-black",
        "textDisabledColor": "#596275",
        "textMonthFontFamily": "libre-bold",
        "textMonthFontSize": 18,
        "textSectionTitleColor": "#cf6a87",
        "todayTextColor": "#cf6a87",
      }
    }
  />
</Styled(View)>
`;
